src/main/java/org/candlepin/pki/certs/SCACertificateGenerator.java

```diff
@@ -58,216 +65,333 @@
+            }
 
-        if (isX509CertExpired) {
             OffsetDateTime start = OffsetDateTime.now().minusHours(1L);
             OffsetDateTime end = start.plusYears(1L);
 
-            KeyPair keyPair = this.keyPairGenerator.getKeyPair(consumer);
-            this.serialCurator.revokeById(existing.getSerial().getId());
-            CertificateSerial serial = createSerial(end);
-            org.candlepin.model.dto.Product container = createSCAProdContainer(owner, consumer);
-            X509Certificate x509Cert = createX509Cert(consumer.getUuid(), owner,
-                serial, keyPair, container, start, end);
+            serial = createSerial(end);
+            KeyPair keypair = keyPairGenerator.getKeyPair(consumer);
+            org.candlepin.model.dto.Product container = createProductContainer(owner, environments);
+
+            X509Certificate x509Cert = createX509Cert(consumer.getUuid(), owner, serial, keypair,
+                container, start, end);
 
-            existing.setSerial(serial);
-            existing.setCert(this.pemEncoder.encodeAsString(x509Cert));
-            this.contentAccessCertificateCurator.saveOrUpdate(existing);
+            scaCertificate.setSerial(serial);
+            scaCertificate.setCert(pemEncoder.encodeAsString(x509Cert));
         }
 
-        Date contentUpdate = owner.getLastContentUpdate();
-        boolean shouldUpdateContent = !contentUpdate.before(existing.getUpdated());
-        if (shouldUpdateContent || isX509CertExpired) {
-            List<Environment> environments = this.environmentCurator.getConsumerEnvironments(consumer);
-            ContentPathBuilder contentPathBuilder = ContentPathBuilder.from(owner, environments);
-            PromotedContent promotedContent = new PromotedContent(contentPathBuilder).withAll(environments);
+        return scaCertificate;
+    }
 
-            Function<ProductContent, String> cidFetcher = pcinfo -> pcinfo.getContent().getId();
+    private ContentAccessPayload getContentAccessPayload(Owner owner, Consumer consumer,
+        List<Environment> environments) throws ConcurrentCertificateCreationException {
 
-            Map<String, ProductContent> ownerContent = this.contentCurator
-                .getActiveContentByOwner(owner.getId())
-                .stream()
-                .collect(Collectors.toMap(cidFetcher, Function.identity(),
-                    (v1, v2) -> new ProductContent(v2.getContent(), v1.isEnabled() || v2.isEnabled())));
+        String payloadKey = new ContentAccessPayloadKeyBuilder(consumer)
+            .setEnvironments(environments)
+            .build();
 
-            byte[] payloadBytes = createContentAccessDataPayload(consumer, ownerContent, promotedContent);
-            existing.setContent(this.createPayloadAndSignature(payloadBytes));
-            this.contentAccessCertificateCurator.saveOrUpdate(existing);
+        ContentAccessPayload container = contentAccessPayloadCurator
```

Comment:
Can we rename container to payload? container within the context of SCA cert generation has been typically the name we use for the mock product that contains the full list of content

Reply:
Updated